components/raftstore/src/store/fsm/peer.rs

```diff
@@ -627,6 +627,7 @@ where
         let count = msgs.len();
         #[allow(const_evaluatable_unchecked)]
         let mut distribution = [0; PeerMsg::<EK>::COUNT];
+        let detail = msgs.get(0).map_or("".to_string(), |m| format!("{:?}", m));
```

Comment:
Just one concern is that the performance.

With this pr, each `msgs[0]` will be checked and formatted with string for the later logging by `slow_log!`. If there exists numerous peers in this node, the formatting cost of `msgs[0]` might be significant.

Reply:
I think you can just pass a `PeerMsg` type to slow_log since it implements the `Debug` trait. The price of close a struct should be much smaller that format it to string while in most case this formatted string is not expected to be used.

Reply:
@glorv Can't as the msg is moved

Reply:
Why clone should be much smaller?

Reply:
![7fbbc528-afcd-44cf-aa1a-9c352923d2e9](https://github.com/tikv/tikv/assets/13497871/b782e88d-7972-4e4a-aec7-5ce2b9657f40)
Let it format only there is only one msg. And the sysbench result turns out to be no impact.

Reply:
PTAL again @glorv @LykxSassinator